### ISSUE TITLE
Version 1.7.0 breaks Stack Navigation when using map Markers

### ISSUE THREAD

**Issue by stjasink** [react-native-maps/react-native-maps#4706]
### Summary

When using `react-native-maps` version `1.7.0`, navigating back from a Stack Navigator screen that contains a MapView with Markers, the app crashes with error:
```com.google.android.gms.maps.GoogleMap cannot be cast to com.google.maps.android.collections.MarkerManager$Collection```

This works fine in version `1.6.0`.

<img width="406" alt="Screenshot 2023-04-23 at 12 29 26" src="https://user-images.githubusercontent.com/8123071/233837751-41431e25-b096-47b2-a594-52c1bf756a95.png">

### Reproducible sample code

```TSX
import React from 'react';
import {Pressable, StyleSheet, Text, View} from 'react-native';
import MapView, {Marker, PROVIDER_GOOGLE} from 'react-native-maps';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';

const Stack = createNativeStackNavigator();

function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="Home">
        <Stack.Screen
          name="Home"
          component={Home}
          options={{title: 'Home Screen'}}
        />
        <Stack.Screen
          name="MyMap"
          component={MyMap}
          options={{title: 'My Map'}}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

function Home({navigation}) {
  return (
    <View>
      <Text>Welcome to the home screen</Text>
      <Pressable onPress={() => navigation.navigate('MyMap')}>
        <Text>Press here for the map</Text>
      </Pressable>
    </View>
  );
}

function MyMap() {
  return (
    <View style={styles.container}>
      <MapView
        provider={PROVIDER_GOOGLE}
        style={styles.map}
        region={mapRegion}
        showsBuildings={false}
        loadingEnabled={true}
        showsUserLocation={true}
        userLocationPriority={'high'}
        showsMyLocationButton={false}
        minZoomLevel={16}
        maxZoomLevel={22}>
        <MyMapMarkers />
      </MapView>
    </View>
  );
}

function MyMapMarkers() {
  return markerLocations.map((location, markerNum) => (
    <Marker
      key={'marker' + markerNum}
      coordinate={{latitude: location.lat, longitude: location.lng}}
      zIndex={markerNum}
      pinColor={'green'}
      trackViewChanges={false}
    />
  ));
}

const styles = StyleSheet.create({
  container: {
    ...StyleSheet.absoluteFillObject,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  map: {
    ...StyleSheet.absoluteFillObject,
  },
});

const mapRegion = {
  latitude: 51.5827395,
  latitudeDelta: 0.0066319,
  longitude: -0.149409,
  longitudeDelta: 0.0019558,
};

const markerLocations = [
  {
    lat: 51.585737,
    lng: -0.150214,
  },
  {
    lat: 51.585365,
    lng: -0.150182,
  },
];

export default App;
```

### Steps to reproduce

Create a new react-native app.
`npm install react-native-maps @react-navigation/native @react-navigation/native-stack react-native-safe-area-context react-native-screens`
Ensure that react-native-maps is at version `1.7.0`
Paste sample code into `App.js`
`npx react-native run-android`

When the app starts up, click on the "Press here for the map" text.  
When the map has rendered, click the back button in the header.
The error will display.

To prove that this is a `1.7.0` bug, change the version to `1.6.0` and re-run.  Clicking the back button will work as expected and display the Home screen.

### Expected result

I expect clicking the back button to navigate back to the Home screen.

### Actual result

An error message is displayed.

<img width="406" alt="Screenshot 2023-04-23 at 12 29 26" src="https://user-images.githubusercontent.com/8123071/233838055-1be03892-5b98-489b-bb68-23163a11e569.png">

### React Native Maps Version

1.7.0

### What platforms are you seeing the problem on?

Android

### React Native Version

0.71.7

### What version of Expo are you using?

Not using Expo

### Device(s)

'Nexus_5X_API_31(AVD) - 12'

**Comment by Simon-TechForm** [comment]
Hi @stjasink

I've just released a fix version that should resolve the issue. Could you try it out and get back to me with the result? :)

**Comment by stjasink** [comment]
> Hi @stjasink
> 
> I've just released a fix version that should resolve the issue. Could you try it out and get back to me with the result? :)

Hi @Simon-TechForm the fix in 1.7.1 worked, thank you! 🎉
